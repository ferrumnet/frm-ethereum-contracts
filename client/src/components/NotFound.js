import React, { Component } from 'react';

export class NotFound extends Component{

    render() {
        return(
            <h1>
                Oooops! Page Not Found
            </h1>
        );
    }
}