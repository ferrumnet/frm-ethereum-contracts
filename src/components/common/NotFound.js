import React from "react";
import { NavLink } from "react-router-dom";
const NotFound = ({minPage = false, message="Oooops! Page Not Found"}) => (
  <div className={`center-content text-center ${minPage?"component-height":"min-component-height"}`}>
    <div>
      <h1>{message}</h1>
      <br />
      <NavLink to="/" className="btn btn-primary">
        {" "}
        Go Back
      </NavLink>
    </div>
  </div>
);

export default NotFound;
