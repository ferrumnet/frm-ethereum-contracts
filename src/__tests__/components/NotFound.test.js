import NotFound from "../../components/common/NotFound";
import { shallow } from "enzyme";
import React from "react";

describe("Test NotFound Component", () => {
  test("It should match the snapshot", () => {
    const wrapper = shallow(<NotFound />);
    expect(wrapper).toMatchSnapshot();
  });
});
