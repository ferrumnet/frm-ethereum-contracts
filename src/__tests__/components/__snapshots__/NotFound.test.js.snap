// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test NotFound Component It should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NotFound />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Oooops! Page Not Found",
    },
    "ref": null,
    "rendered": "Oooops! Page Not Found",
    "type": "h1",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Oooops! Page Not Found",
      },
      "ref": null,
      "rendered": "Oooops! Page Not Found",
      "type": "h1",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
